### PR TITLE
Update EIA Fuel Mix for New Power Storage Columns

### PR DESCRIPTION
**Pull request by WillKoehrsen**
## Summary

- Updates `EIA()` fuel mix retrieval to handle new battery storage columns

### CODE REVIEW

File: gridstatus/eia.py [hunk 1]
```diff
@@ -741,14 +755,49 @@ def _handle_fuel_type_data(df):
...
+        if col not in df.columns:
+            # This has to be np.nan not pd.NA because we are converting to float
+            df[col] = np.nan
+
     fuel_mix_cols = df.columns[4:]
 
-    # nans after pivot because not
-    # all respondents have all fuel types
+    # nans after pivot because not all respondents have all fuel types
+    # Why are we filling these with 0? A value of 0 means something different than NaN
```
Comment (WillKoehrsen) [inline gridstatus/eia.py:793]: @kmax12 Why are the na values being filled with 0? I would think they should be kept as na to avoid creating data values.

Comment (kmax12) [reply gridstatus/eia.py:801]: i think my logic was that if the fuel didn't get reported, it means it wasn't generated, hence it is okay to say zero. 

I'm okay if you want to change it to null.

if you do, we'll want to put in the description of the dataset that null means that it wasn't reported.

**Review by kmax12 — APPROVED** [review]

File: gridstatus/eia.py [hunk 2]
```diff
@@ -734,20 +753,62 @@ def _handle_fuel_type_data(df):
...
+    # (e.g. "Pumped Storage","Pumped storage"). We can handle that through the pivot
+    # table by summing the duplicates.
+    df["type-name"] = df["type-name"].str.lower()
+
     # pivot on type
     df = df.pivot_table(
         index=["Interval Start", "Interval End", "Respondent", "Respondent Name"],
         columns="type-name",
         values="MW",
+        aggfunc="sum",
```
Comment (WillKoehrsen) [inline gridstatus/eia.py:766]: This assumes that "Pumped storage" and "Pumped Storage" can be added together if they are both present.

File: gridstatus/eia.py [hunk 3]
```diff
@@ -734,20 +753,62 @@ def _handle_fuel_type_data(df):
...
-    df.columns.name = None
+    for col in known_fuel_mix_columns:
+        if col not in df.columns:
+            # This has to be np.nan not pd.NA because we are converting to float
+            df[col] = np.nan
+
+    fixed_cols = ["Interval Start", "Interval End", "Respondent", "Respondent Name"]
+    fuel_mix_cols = [col for col in df.columns if col not in fixed_cols]
+
+    df[fuel_mix_cols] = df[fuel_mix_cols].astype(float)
```
Comment (WillKoehrsen) [inline gridstatus/eia.py:802]: Updated to not fill in NAs with 0.

File: gridstatus/eia.py [hunk 4]
```diff
@@ -734,20 +753,69 @@ def _handle_fuel_type_data(df):
...
+    # table by summing the duplicates.
+    df["type-name"] = df["type-name"].str.lower()
+
+    # These columns are grouped together by EIA as confirmed by inspection of the EIA
+    # fuel mix graphs
+    df["type-name"] = df["type-name"].replace(
+        {
+            "battery": "battery storage",
+            "solar battery": "solar with integrated battery storage",
+            "unknown energy": "unknown energy storage",
```
Comment (WillKoehrsen) [inline gridstatus/eia.py:767]: These combinations are what EIA does on their charts. It's possible (likely?) EIA will change these again or start publishing additional fuel types.

https://www.eia.gov/electricity/gridmonitor/expanded-view/electric_overview/US48/US48/GenerationByEnergySource-4/edit

Comment (kmax12) [reply gridstatus/eia.py:767]: i think you can leave this comment in the code too

**Review by kmax12 — APPROVED** [review]

**Pull request merged**